CMakeLists.txt

```diff
@@ -2029,6 +2085,52 @@ target_include_directories(cocos2d PUBLIC
     ${CWD}/external/sources/khronos
 )
 
+if(NINTENDO)
+    target_compile_definitions(cocos2d PUBLIC CC_USE_NVN)
+
+    if(${CMAKE_SIZEOF_VOID_P} STREQUAL "4")
+        if(${MSVC_VERSION} LESS 1920)
+            set(NN_SDK_PLATFORM_FOLDER win32-v141)
+        else()
+            set(NN_SDK_PLATFORM_FOLDER win32-v142)
+        endif()
+    else()
+        if(${MSVC_VERSION} LESS 1920)
+            set(NN_SDK_PLATFORM_FOLDER x64-v141)
+        else()
+            set(NN_SDK_PLATFORM_FOLDER x64-v142)
+        endif()
+    endif()
+
+    if(CC_PLATFORM EQUAL ${CC_PLATFORM_NINTENDO})
+        set(NN_SDK_PLATFORM_FOLDER NX-${NN_SDK_PLATFORM_FOLDER})
+    endif()
+
+    target_include_directories(cocos2d PUBLIC
+        $ENV{NINTENDO_SDK_ROOT}/Include
+        $ENV{NINTENDO_SDK_ROOT}/Common/Configs/Targets/${NN_SDK_PLATFORM_FOLDER}/Include
+    )
+
+    add_library(NN_SDK STATIC IMPORTED GLOBAL)
+    set_target_properties(NN_SDK PROPERTIES
+      IMPORTED_LOCATION_DEBUG $ENV{NINTENDO_SDK_ROOT}/Libraries/${NN_SDK_PLATFORM_FOLDER}/Debug/libnnSdk.lib
+      IMPORTED_LOCATION_RELEASE $ENV{NINTENDO_SDK_ROOT}/Libraries/${NN_SDK_PLATFORM_FOLDER}/Release/libnnSdk.lib
+    )
+
+    add_library(NN_SDK_DIAG STATIC IMPORTED GLOBAL)
+    set_target_properties(NN_SDK_DIAG PROPERTIES
+      IMPORTED_LOCATION_DEBUG $ENV{NINTENDO_SDK_ROOT}/Libraries/${NN_SDK_PLATFORM_FOLDER}/Debug/libnn_diagTextPublicEn.lib
+      IMPORTED_LOCATION_RELEASE $ENV{NINTENDO_SDK_ROOT}/Libraries/${NN_SDK_PLATFORM_FOLDER}/Release/libnn_diagTextPublicEn.lib
+    )
+
+    add_library(NVN_LOADER STATIC IMPORTED GLOBAL)
```

Comment:
`NINTENDO_SDK_LOADER`

Reply:
Here too. I think the name `NVN` is already (sort of) in the public domain...? Can we get at least this much leeway?

Reply:
It's simply breaking the agreement if judged. If I see the risk, I can't ignore it